homeassistant/components/mqtt/mixins.py

```diff
@@ -337,6 +339,118 @@ async def _async_setup_entities() -> None:
     await _async_setup_entities()
 
 
+async def async_mqtt_entry_helper(
```

Comment:
If `async_setup_entry_helper` will not be removed in the end, please refactor `async_setup_entry_helper` and `async_mqtt_entry_helper` to avoid the duplication of code.
Also, please give them better names, I guess one is for setting up entities and one is for setting up non entities after these changes?

Reply:
Agree, but can we do this at the end of migrating the platforms as it would spare a lot work.

Reply:
I'll add a commit to de-duplicate the code of the two helpers in #101649

Reply:
https://github.com/home-assistant/core/pull/101649/commits/2e047b2dbc0829aa5e983bdf0592579fcd302ba8